File: c/src/Behavioral/Visitor/GameInfo.c

//GameInfo - three of three concrete Visitees

#include "TitleInfo.h"
#include "GameInfo.h"

#include "stdlib.h"

void GameInfo_accept(void * this, TitleBlurbVisitor_t * tbv) 
{
	tbv->visit(tbv,this);
}

TitleInfo_t * GameInfo_new(char * titleName)
{
	TitleInfo_t * bi = TitleInfo_new( GameInfo_accept, titleName );
	bi->type = GameInfoType;
	return bi;
}    
   
void GameInfo_free(TitleInfo_t * d)
{
   if(d == NULL)
      return;
   free(d);
}

File: go/behavioral/visitor/GameInfo.go
package visitor

import (
	"fmt"
)


type GameInfo struct { // implements Acceptor
	Title
}

func NewGameInfo(titleName string ) *GameInfo {
	return &GameInfo{&TitleInfo{titleName}}
}    

func (this *GameInfo) Accept( tbv Visitor ) {
	tbv.Visit( this )
}

func (this *GameInfo) GetLongBlurb() string {
	return fmt.Sprintf("LB-Game: %s" ,this.GetTitle() )
}

func (this *GameInfo) GetShortBlurb() string {
	return fmt.Sprintf("SB-Game: %s" , this.GetTitle() )
}
